pkgs/servers/nextcloud/packages/apps/recognize.nix

```diff
@@ -0,0 +1,136 @@
+    };
+    "29" = {
+      version = "7.1.0";
+      appHash = "sha256-qR4SrTHFAc4YWiZAsL94XcH4VZqYtkRLa0y+NdiFZus=";
+      modelHash = "sha256-M/j5wVOBLR7xMVJQWDUWAzLajRUBYEzHSNBsRSBUgfM=";
+    };
+    "28" = {
+      # Once this version is no longer supported, we can remove the getAppValue replacements below
+      # The getAppValueString stuff will need to remain
+      version = "6.1.0";
+      appHash = "sha256-225r2JnDOoURvLmzpmHp/QL6GDx9124/YTywbxH3/rk=";
+      modelHash = "sha256-4mhQM/ajpwjqTb8jSbEIdtSRrWZEOaMZQXAwcfSRQ/M=";
+    };
+  };
+  currentVersionInfo = latestVersionForNc.${ncVersion};
+in
+stdenv.mkDerivation rec {
+
+  pname = "nextcloud-app-recognise";
+  version = currentVersionInfo.version;
+
+  srcs =
+    [
+      (fetchurl {
+        inherit version;
+        url = "https://github.com/nextcloud/recognize/releases/download/v${version}/recognize-${version}.tar.gz";
+        hash = currentVersionInfo.appHash;
+      })
+
+      (fetchurl {
+        inherit version;
+        url = "https://github.com/nextcloud/recognize/archive/refs/tags/v${version}.tar.gz";
+        hash = currentVersionInfo.modelHash;
+      })
+    ]
+    ++ lib.optionals useLibTensorflow [
+      (fetchurl rec {
+        # For version see LIBTENSORFLOW_VERSION in https://github.com/tensorflow/tfjs/blob/master/tfjs-node/scripts/deps-constants.js
+        version = "2.9.1";
+        url = "https://storage.googleapis.com/tensorflow/libtensorflow/libtensorflow-cpu-linux-x86_64-${version}.tar.gz";
+        hash = "sha256-f1ENJUbj214QsdEZRjaJAD1YeEKJKtPJW8pRz4KCAXM=";
+      })
+    ];
+
+  unpackPhase =
+    ''
+      # Merge the app and the models from github
+      tar -xzpf "${builtins.elemAt srcs 0}" recognize;
+      tar -xzpf "${builtins.elemAt srcs 1}" -C recognize --strip-components=1 recognize-${version}/models
+    ''
+    + lib.optionalString useLibTensorflow ''
+      # Place the tensorflow lib at the right place for building
+      tar -xzpf "${builtins.elemAt srcs 2}" -C recognize/node_modules/@tensorflow/tfjs-node/deps
+    '';
+
+  patchPhase = ''
+    # Make it clear we are not reading the node in settings
+    sed -i "/'node_binary'/s:'""':'Nix Controled':" recognize/lib/Service/SettingsService.php
+
+    # Replace all occurences of node (and check that we actually remved them all)
+    test "$(grep "get[a-zA-Z]*('node_binary'" recognize/lib/**/*.php | wc -l)" -gt 0
+    substituteInPlace recognize/lib/**/*.php \
+      --replace-quiet "\$this->settingsService->getSetting('node_binary')" "'${nodejs}/bin/node'" \
+      --replace-quiet "\$this->config->getAppValueString('node_binary', '""')" "'${nodejs}/bin/node'" \
+      --replace-quiet "\$this->config->getAppValueString('node_binary')" "'${nodejs}/bin/node'" \
+      --replace-quiet "\$this->config->getAppValue('node_binary', '""')" "'${nodejs}/bin/node'" \
+      --replace-quiet "\$this->config->getAppValue('node_binary')" "'${nodejs}/bin/node'"
```

Comment:
```suggestion
      --replace-error "\$this->settingsService->getSetting('node_binary')" "'${nodejs}/bin/node'" \
      --replace-error "\$this->config->getAppValueString('node_binary', '""')" "'${nodejs}/bin/node'" \
      --replace-error "\$this->config->getAppValueString('node_binary')" "'${nodejs}/bin/node'" \
      --replace-error "\$this->config->getAppValue('node_binary', '""')" "'${nodejs}/bin/node'" \
      --replace-error "\$this->config->getAppValue('node_binary')" "'${nodejs}/bin/node'"
```

Reply:
Indeed.

There is a test verifying the replacement, not all files need replacing, bu by doing it this way it should keep working for longer